search/api.py

```diff
@@ -336,7 +336,9 @@ def update_percolate_memberships(user, source_type):
         user
     )
 
-    _update_memberships([query.id for query in percolate_queries], membership_ids, user)
+    # if there are no percolate queries or memberships then there's nothing to do
+    if percolate_queries and membership_ids:
```

Comment:
This should be an `or` I think. If only one or the other is false it seems like we want to do the update, right?

Reply:
No, if `membership_ids` is empty, there are no memberships to process, if `percolate_queries` is empty, there's also nothing to do, although arguably, if `percolate_queries` is empty, `membership_ids` will be too as we would have not created any on line 334. So really, in theory, just checking one should be sufficient.

Either that or I could return earlier if `percolate_queries` is empty, but I think the difference between the two approaches is more or less moot.

Reply:
If it doesn't matter which one we check, can we just check `membership_ids` to make this logic simpler?